python/tester/utils.py

```diff
@@ -216,8 +217,11 @@ def make_lsst_id(group_id, snap, abbrev):
     # controller in the header.
     controller = "O"
     obs_id = f"{abbrev}_{controller}_{day_obs}_{seq_num:06d}"
+    mjd = Time.strptime(day_obs, "%Y%m%d").mjd
     return exposure_num, {
         "DAYOBS": day_obs,
+        "MJD-OBS": mjd,
+        "DATE-END": None,
```

Comment:
I guess I'm still confused about why we didn't have a problem with earlier LATISS and ComCamSim uploads. `day_obs` and MJD are both defined without reference to any country's summer time (let alone USA's), so where does the variable offset come from?

Reply:
My understanding is that the MJD determines the observation start time, and that determines an observing date offset which may or may not have daylight saving.  Before the camera is shipped to Chile (`_LSST_CAM_SHIP_DATE`)  it [follows Pacific time](  https://github.com/lsst/obs_lsst/blob/w.2024.16/python/lsst/obs/lsst/translators/lsstCam.py#L186) with daylight saving, so has a different offset when MJD is in summer or in winter. 

We did not test LATISS nor ComCamSim uploads at scale in `dev` before, and our previous tests read from small datasets; by chance our current `unobserved` folder has only LATISS raws from summer time, and only ComCamSim raws from winter time. 

Also the `day_obs` table is new in Butler. Previously we didn't bother to modify MJD in the upload tests; it might fetch the calibration of the original MJD but we don't care. 

I first saw the conflict when I mixed, on one same day, the winter time ComCamSim data from the `unobserved` folder and the summer time ops-rehearsal-3 ComCamSim data from `/repo/embargo`.  They both got the modified `DAYOBS` to be "today", but one uses the winter time offset and the other uses the summer time offset, hence the conflict in butler.  

There is also [DM-43495](https://rubinobs.atlassian.net/browse/DM-43495)  though that's not the source of our problem here.

Reply:
[DM-43495](https://rubinobs.atlassian.net/browse/DM-43495) is actually very relevant too. 

With DM-43495, it's no longer necessary to modify `MJD-OBS` and `DATE-END` / mock an observation date, because all data Prompt Processing cares have one same offset (post-DM-43495), instead of a daylight-saving-dependent offset (pre-DM-43495).   So no more conflicts from raws between summer versus winter time.  The original observing dates can be used to get calibrations, just like how our HSC dev test works. 

I'll remove this commit (https://github.com/lsst-dm/prompt_processing/pull/154/commits/f0129dd848ba10b77bde5b68d75f39442236da9e).    Using the original observation date also means that LATISS tests can run into the same calib validity range conflicts just like HSC ([DM-43205](https://rubinobs.atlassian.net/browse/DM-43205)). 

Just like HSC-dev, I'll turn off calib-caching for LATISS-dev.  Alternatively, we leave it on, and when we run larger LATISS tests in `dev` we can turn if off temporarily.  

LSSTComCamSim can still have calib-caching, because there is only one version of calibrations so far.